setup/compiler.class.inc.php

```diff
@@ -3221,10 +3222,11 @@ protected function CompileDictionaries($oDictionaries, $sTempTargetDir, $sFinalT
 
 			$aEntriesPHP = array();
 			$oEntries = $oDictionaryNode->GetUniqueElement('entries');
+			/** @var MFElement $oEntry */
 			foreach ($oEntries->getElementsByTagName('entry') as $oEntry)
 			{
 				$sStringCode = $oEntry->getAttribute('id');
-				$sValue = $oEntry->GetText();
+				$sValue = $oEntry->GetText('');
 				$aEntriesPHP[] = "\t'$sStringCode' => ".self::QuoteForPHP(self::FilterDictString($sValue), true).",";
```

Comment:
```suggestion
				$sValue = $oEntry->GetText();
				$aEntriesPHP[] = "\t'$sStringCode' => ".self::QuoteForPHP(self::FilterDictString($sValue ?? ''), true).",";
```

See last comment in the conversation

Reply:
Hmm, I thought "We would rather fix the root cause than the symptom" would mean you wanted the root cause fixed instead of adding a workaround?

Reply:
Yes, what we meant by that is that the `FilterDictString` should be stricter to avoid this symptom. But I overlooked your proposition, it's actually ok. Thanks.